### PR TITLE
Add search tab to editor

### LINKED ISSUE #583: Search Tab -  Simple Search and Results Table
Conferred with @astridu regarding the UI for the **Browse** tab in the Editor. Here is an earlier design that we can use (with red x on parts of the design we are no longer using):

Screen Shot 2019-05-29 at 3 56 19 PM

This would have a single input field to do a keyword search in Sinopia's Elasticsearch index and return a list of entities that are ordered by relevancy.

### PR DESCRIPTION
**Pull request by aaron-collier**
Closes #583

This replaces the browse tab with a search tab that allows for searching against elasticsearch. A non-paged list of resources is displayed (if any found) and when clicking on a result it loads the resource into the editor.

### CODE REVIEW

File: __tests__/components/Header.test.js [hunk 1]
```diff
@@ -25,8 +25,11 @@ describe('<Header />', () => {
 
   describe('nav tabs', () => {
     describe('when no resource in state', () => {
-      it('displays 3 header tabs', () => {
-        expect(wrapper.find('ul.editor-navtabs NavLink').length).toBe(3)
+      it('displays 4 header tabs', () => {
```
Comment (jcoyne) [inline __tests__/components/Header.test.js]: I thought the Browse tab is the same as the search tab, not a new tab.  Currently the browse tab indicates:
> This is a temporary facility until we have search implemented.

File: src/reducers/index.js [hunk 2]
```diff
@@ -3,8 +3,8 @@
 import { combineReducers } from 'redux'
 import authenticate from './authenticate'
 import {
-  removeMyItem, setItemsOrSelections, setBaseURL, showResourceURIMessage, setMyItemsLang,
-  showGroupChooser, closeGroupChooser, showRdfPreview,
+  removeAllContent, removeMyItem, setItemsOrSelections, setBaseURL, showResourceURIMessage, setMyItemsLang,
```
Comment (jcoyne) [inline src/reducers/index.js]: Why are you adding removeAllContent back?

File: src/components/search/SearchResults.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import Button from 'react-bootstrap/lib/Button'
+import ButtonToolbar from 'react-bootstrap/lib/ButtonToolbar'
+
+const SearchResults = (props) => {
+  const handleClick = resourceURI => props.loadResource(props.currentUser, resourceURI)
+
+  // This returns the current row number + 1 in order to include it in the displayed table
+  const indexFormatter = (_cell, _row, rowIndex, _formatExtraData) => rowIndex + 1
+
+  // const titleFormatter = (_cell, row, _rowIndex, _formatExtraData) => <a href="http://localhost:8888">{ row.title }</a>
```
Comment (jcoyne) [inline src/components/search/SearchResults.jsx]: cruft?

File: src/reducers/index.js [hunk 4]
```diff
@@ -133,11 +133,13 @@ const handlers = {
   RETRIEVE_RESOURCE_TEMPLATE_STARTED: clearRetrieveError,
   SET_BASE_URL: setBaseURL,
   SHOW_RESOURCE_URI_MESSAGE: showResourceURIMessage,
+  SHOW_SEARCH_RESULTS: showSearchResults,
   SHOW_GROUP_CHOOSER: showGroupChooser,
   CLOSE_GROUP_CHOOSER: closeGroupChooser,
   LANGUAGE_SELECTED: setMyItemsLang,
   SHOW_RDF_PREVIEW: showRdfPreview,
   REMOVE_ITEM: removeMyItem,
+  REMOVE_ALL_CONTENT: removeAllContent,
```
Comment (jcoyne) [inline src/reducers/index.js]: Is this intentional?

File: __tests__/components/Header.test.js (hunk 1 above)
Comment (aaron-collier) [reply __tests__/components/Header.test.js]: I can remove browse. The mock up had "Search", so I didn't think we'd want to replace it yet, but I'm fine with that.

File: src/reducers/index.js (hunk 2 above)
Comment (aaron-collier) [reply src/reducers/index.js]: Cruft, that shouldn't have stayed there. Will fix.

File: src/components/search/SearchResults.jsx (hunk 3 above)
Comment (aaron-collier) [reply src/components/search/SearchResults.jsx]: yup. will fix. thanks.

File: src/reducers/index.js (hunk 4 above)
Comment (aaron-collier) [reply src/reducers/index.js]: No. I'll clean up, thanks.

File: package.json [hunk 5]
```diff
@@ -83,8 +83,8 @@
     "dev-build": "webpack --progress --mode development",
     "dev-build-test": "npm run dev-build && npm run test",
     "dev-start": "webpack-dev-server --config ./webpack.config.js --mode development",
-    "eslint": "eslint --max-warnings 5 --color -c .eslintrc.js --ext js,jsx .",
-    "eslint-fix": "eslint --max-warnings 5 --color -c .eslintrc.js --ext js,jsx --fix .",
+    "eslint": "eslint --max-warnings 6 --color -c .eslintrc.js --ext js,jsx .",
```
Comment (jcoyne) [inline package.json]: Can you add the ignore pragma to the file rather than just raising the threshold?

Comment (aaron-collier) [reply package.json]: Set it back to 5. The extra warning I was getting wasn't in any of my code, but appears to be removed after rebasing... so no change was needed here.

**Pull request merged**